optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/stream/BavetConstraintStreamScoreDirector.java

```diff
@@ -37,12 +37,13 @@
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  * @see ScoreDirector
  */
-public class ConstraintStreamScoreDirector<Solution_, Score_ extends Score<Score_>>
-        extends AbstractScoreDirector<Solution_, Score_, AbstractConstraintStreamScoreDirectorFactory<Solution_, Score_>> {
+public final class BavetConstraintStreamScoreDirector<Solution_, Score_ extends Score<Score_>>
+        extends AbstractScoreDirector<Solution_, Score_, BavetConstraintStreamScoreDirectorFactory<Solution_, Score_>> {
 
-    protected ConstraintSession<Solution_, Score_> session;
+    protected BavetConstraintSession<Solution_, Score_> session;
```

Comment:
Can't we roll the BavetConstraintSession also into BavetConstraintStreamScoreDirector?

Reply:
Ok, maybe for a later PR.